kvclient.go

```diff
@@ -202,33 +209,60 @@ func NewKvClient(ctx context.Context, opts *KvClientConfig) (*kvClient, error) {
 	return kvCli, nil
 }
 
-func (c *kvClient) Reconfigure(ctx context.Context, opts *KvClientConfig) error {
-	if opts == nil {
-		return nil
+func (c *kvClient) Reconfigure(config *KvClientConfig, cb func(error)) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if config == nil {
```

Comment:
This should probably be before the lock, but that's being picky.

Reply:
Fixed.